src/uct/base/uct_iface.c

```diff
@@ -202,6 +202,72 @@ static ucs_status_t uct_base_ep_flush(uct_ep_h tl_ep, unsigned flags,
     return UCS_OK;
 }
 
+static void uct_ep_failed_purge(uct_ep_h ep, uct_pending_callback_t cb)
+{
+    uct_failed_iface_t *iface = ucs_derived_of(ep->iface, uct_failed_iface_t);
+    uct_pending_req_t *req;
+
+    ucs_queue_for_each_extract(req, &iface->pend_q, priv, 1) {
+        if (cb != NULL) {
```

Comment:
Not sure. It is not described in the doxygen docs and I've seen some other transports do that (mostly for internal cleanup). I was going (but forgot) to update RC endpoint  cleanup routine to  print warnings instead of failing with assertion. So it'd be possible to just call purge with callback=NULL in ep destructor. Will update that shortly